src/redux/books.js/booksSlice.js

```diff
@@ -30,11 +30,7 @@ const booksSlice = createSlice({
       state.bookItem.push(action.payload);
     },
     removeBook: (state, action) => {
-      const bookId = action.payload;
-      return {
-        ...state,
-        bookIt25056em: state.bookItem.filter((book) => book.id !== bookId),
-      };
+      state.books = state.books.filter((book) => book.id !== action.id);
     },
```

Comment:
- [ ] When I try to delete a book, I get this error:

![image](https://github.com/techmoves/bookstores/assets/77106962/a5bbeb7a-646a-4907-b43e-0b5c58103496)

There is no `state.books` variable/object. Remember that initialState contains an object named `bookItem` which contains an array of books. Therefore, to access it, you need to make your statement like this:

```JavaScript
state.bookItem = state.bookItem.filter((book) => book.item_id !== action.payload);
```

Please ensure that the application can delete a particular book one at a time. 👍 💻